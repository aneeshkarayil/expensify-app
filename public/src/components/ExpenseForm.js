import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: ""
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = e => {
    const amount = e.target.value;

    if (amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({
        createdAt
      }));
    }
  };

  onFocusChange = focused => {
    console.log(focused);
    this.setState(() => ({
      calendarFocused: focused
    }));
    console.log(this.state);
  };

  onSubmit = e => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      this.setState(() => {
        return {
          error: "Please set the description and amount"
        };
      });
    } else {
      this.setState(() => {
        return {
          error: ""
        };
      });
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <div>
        <h3>{this.state.error}</h3>
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          ></input>
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          ></input>
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            id="date" // Here, need to add id
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            placeholder="Add a note for the expense"
            value={this.state.note}
            onChange={this.onNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
