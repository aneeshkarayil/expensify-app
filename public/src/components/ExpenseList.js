import React from "react";
import { connect } from "react-redux";
import ExpenseLineItem from "./ExpenseListItem";
import selectExpense from "../selectors/expenses";

export const ExpenseList = props => (
  <div>
    <h1>Expense List</h1>
    {props.expenses.map(expense => {
      return <ExpenseLineItem key={expense.id} {...expense} />;
    })}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpense(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
