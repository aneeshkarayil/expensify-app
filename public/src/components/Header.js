import React from "react";
import { Switch, Route, BrowserRouter, Link, NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";
import { connect } from "react-redux";

export const Header = ({ startLogout }) => (
  <div>
    <header className="header">
      <div className="content-container">
        <div className="header__content">
          <Link to="/dashboard" exact={true} className="header__title">
            <h1>Expensify</h1>
          </Link>
          <button onClick={startLogout} className="button button--link">
            Log out
          </button>
        </div>
      </div>
    </header>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
