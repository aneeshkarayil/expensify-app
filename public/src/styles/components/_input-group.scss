.input-group {
  display: flex;
  flex-direction: row;
  margin-bottom: $m-size;

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin-bottom: $l-size;
  }
}

.input-group__item {
  margin-bottom: $s-size;
  margin-right: $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin: 0 $s-size 0;
  }
}
