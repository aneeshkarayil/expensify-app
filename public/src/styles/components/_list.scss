.list-header {
  background: #f7f7f7;
  border: 1px solid darken(#f7f7f7, 7%);
  color: grey;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-item {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border: 1px solid darken(#f7f7f7, 7%);
  border-top: none;
  color: $dark-grey;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: #f7f7f7;
  }
}

.list-item__title {
  margin: 0;
  word-break: break-all;
}

.list-item__sub-title {
  color: $dark-grey;
  font-size: $font-size-small;
}

.list-item__data {
  margin: 0;
  padding-left: $s-size;
}
