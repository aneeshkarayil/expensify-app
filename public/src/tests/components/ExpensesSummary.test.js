import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("Should correctly render expenses summary with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={201.2} />
  );

  expect(wrapper).toMatchSnapshot();
});

test("Should correctly render expenses summary with multiple expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={23} expensesTotal={201.2} />
  );

  expect(wrapper).toMatchSnapshot();
});
