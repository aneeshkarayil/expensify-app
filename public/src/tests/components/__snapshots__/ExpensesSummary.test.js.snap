// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expenses
     totalling 
    $201.20
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $201.20
  </h1>
</div>
`;
